docusaurus/docs/recipes/dropdown.md

```diff
@@ -0,0 +1,427 @@
+---
+id: dropdown
+title: Implementing a Dropdown
+sidebar_label: Dropdown
+---
+
+Implementing a Dropdown/Select component from the scratch is always a big challenge. A lot of things have to be considered being some of them accessibility, autocomplete feature, multi-selection feature, native support (for touch devices mainly).
+
+There are open-source libraries like [react-select](https://github.com/JedWatson/react-select) to help us out on this task but usually their API is not that friendly. This happens because these libraries will render the input and the menu for themselves. For some of them, we are allowed to change what we want to render but it will render on a specific location. Other ones provide a list of `classNames` that can be used to change the style of the rendered component the way it fits best for us.
+
+The main problem of these libraries is the lack of flexibility while using them to build our custom dropdown. That's where [downshift](https://github.com/downshift-js/downshift) comes into play.
+
+### What is Downshift 🏎?
+
+> Primitives to build simple, flexible, WAI-ARIA compliant React autocomplete, combobox or select dropdown components.
+
+This is how [Kent C. Dodds](https://kentcdodds.com/) describes his library.
+
+Downshift mainly takes care of three things: **managing state**; **user interactions**; and **accessibility**. The rendering task is all up to the developer. This is possible because it uses both [render prop pattern](https://reactjs.org/docs/render-props.html) and [prop getters concept](https://kentcdodds.com/blog/how-to-give-rendering-control-to-users-with-prop-getters) as its core concept.
```

Comment:
```suggestion
Downshift mainly takes care of three things: **managing state**, **user interactions** and **accessibility**. The rendering task is all up to the developer. This is possible because it uses both the [render prop pattern](https://reactjs.org/docs/render-props.html) and the [prop getters concept](https://kentcdodds.com/blog/how-to-give-rendering-control-to-users-with-prop-getters) at its core.
```